python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -1182,6 +1183,23 @@ def to_selection_str(self) -> str:
         return f'key_substring:"{self.selected_key_substring}"'
 
 
+@whitelist_for_serdes
+@record
+class KeyWildCardAssetSelection(AssetSelection):
+    selected_key_wildcard: str
```

Comment:
Not really happy with the naming here 🙃

Reply:
Seems OK to me, otherwise maybe `WildcardMatchedKeyAssetSelection` or `KeyWildcardMatchedAssetSelection`?

Reply:
`KeyAssetSelectionWithWildcard` 🙃